shopcart.py

```diff
@@ -75,7 +75,111 @@
 ######################################################################
 @app.route('/')
 def index():
-    return make_response("Hello World", HTTP_200_OK)
+    shopcarts_url = request.base_url + "shopcarts"
+    return make_response(jsonify(name='Shopcart Demo REST API Service',
+                   version='1.0',
+                   url=shopcarts_url
+                   ), HTTP_200_OK)
+
+######################################################################
+# LIST ALL SHOPCARTS FOR ALL USERS
+######################################################################
+#USAGE: /shopcarts or #USAGE: /shopcarts?sid=1
+@app.route('/shopcarts', methods=['GET'])
+def list_shopcarts():
+    results=[]
+    sid=request.args.get('sid')
+    if sid:
+        #Extra check for query input
+        try:
+            sid = int(sid)
+        except ValueError:
+            return make_response(jsonify(shopping_carts), HTTP_200_OK)
+
+        results=[cart for cart in shopping_carts if cart['sid']==int(sid)]
+        if len(results)==0:
+            results=shopping_carts
+        else:
+
+            if len(results[0]['products'])==0:
+                results='The cart contains no products'
+            else:
+                results=results[0]['products']
+    else:
+        results=shopping_carts
+
+    return make_response(jsonify(results), HTTP_200_OK)
+
+######################################################################
+# RETRIEVE USER'S PRODUCTS IN THE CART
+######################################################################
```

Comment:
No I meant the comment at the start of the API since we are retrieving the whole shopping cart, doesnt retrieve user's products in the cart sound misleading. Beside the API below also has the same initial comment.